src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs

```diff
@@ -1521,22 +1521,15 @@ private static bool VectorContainsNonAsciiChar(Vector128<byte> asciiVector)
         internal static bool VectorContainsNonAsciiChar(Vector128<ushort> utf16Vector)
         {
             // prefer architecture specific intrinsic as they offer better perf
-            if (Sse2.IsSupported)
+#pragma warning disable IntrinsicsInSystemPrivateCoreLibConditionParsing // A negated IsSupported condition isn't parseable by the intrinsics analyzer
+            if (Sse2.IsSupported && !Sse41.IsSupported)
```

Comment:
The warning exists because this is going to flag the code as depending on SSE4.1

We should likely not consider the SSE2 path and just have SSE4.1, as such hardware is increasingly irrelevant (more than 17 years old at this point) and even crossgen is defaulting to an opportunistic SSE4.1 target today

Reply:
That is, remove this path entirely and just have `if (AdvSimd.Arm64.IsSupported) { } else { }`

This will do the right thing for SSE4.1 already and while it might slightly pessimize an SSE2 only machine, such machines are old/rare enough that's acceptable IMO. We're talking about what ends up being an instruction or so difference in practice

Reply:
These changes can be made in a follow-up PR.

Reply:
@tannergooding Is d21415f7c26d7e85e3e219279a2aeaa5e567768c still problematic?

Reply:
I believe the root issue should be resolved, but its not really ideal and leads to potentially confusing control flow.

Arguably we don't really need the `Sse2` specific code path (I don't believe anything will even call this method in that scenario today, I believe the callers require Ssse3) and simply `if (AdvSimd.Arm64.IsSupported) { } else { }` is sufficient. -- More ideally we wouldn't need the Arm64 path at all, because this is a relatively trivial pattern that the JIT could optimize itself but that's outside the scope of this PR.

Reply:
Callers only require Sse2

https://github.com/dotnet/runtime/blob/a538054ec46ecf3587fc6fc08603324fdfdb77b2/src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Utility.cs#L712-L723

Reply:
`Vector128.IsHardwareAccelerated` path:

https://github.com/dotnet/runtime/blob/a538054ec46ecf3587fc6fc08603324fdfdb77b2/src/libraries/System.Private.CoreLib/src/System/Text/Ascii.CaseConversion.cs#L236-L241

Reply:
> then we'd want to duplicate the code or have the JIT handling the general pattern itself.

we have the patterns here because of how S.P.Corelib is specially handled by the tooling, it is important we follow them so that things work as expected, so things are trimmed, so AOT and R2R work, etc

Reply:
653aef83b0b3f4cc786d9f442b50dd8032f74076

Reply:
> then we'd want to duplicate the code or have the JIT handling the general pattern itself.

Good example of the Rule of Three for code duplication ("Three strikes and you refactor").